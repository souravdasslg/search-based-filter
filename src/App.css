.center {
  position: relative;
  margin: 1%;
}
.main {
  height: 600px;
  width: 300px;
  background: lightblue;
  border: 3px solid darkblue;
}
.message {
  color: darkblue;
  align-items: center;
  margin: 20px;
}