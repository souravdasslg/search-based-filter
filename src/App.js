import React, { useState } from "react";
import "./App.css";
import SearchComponent from "./components/SearchComponet/SearchComponent";
import { search } from "./lib/searchFilterLib";
import DataViewComponent from "./components/Data Viewer/DataViewerComponent";
function App() {
  const [filterList, setFilterList] = useState([]);
  const [searchInput, setSearchInput] = useState("");
  const handleSearchInput = ({ target }) => {
    const value = target.value;
    const searchResult = search(value);
    setFilterList(searchResult);
    setSearchInput(value);
  };

  return (
    <main className="center main">
      <SearchComponent onChange={handleSearchInput} />
      {searchInput ? (
        <DataViewComponent list={filterList} />
      ) : (
        <div className="message">Nothing to search...</div>
      )}
    </main>
  );
}

export default App;
