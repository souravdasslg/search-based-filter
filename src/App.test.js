import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

test("Match Snapshot For Initial Rendering", () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
