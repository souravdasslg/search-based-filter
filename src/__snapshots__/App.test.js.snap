// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot For Initial Rendering 1`] = `
<div>
  <main
    class="center main"
  >
    <div>
      <input
        aria-label="Input label to search"
        class="search_input"
        placeholder="Type your keyword here"
      />
    </div>
    <div
      class="message"
    >
      "Start searching filters by typings"
    </div>
  </main>
</div>
`;
